### DISCUSSION TITLE
why "make -j4" command has bellow error, help?

### DISCUSSION THREAD

**Discussion by Dawn369** [microsoft/msquic#3456] [Q&A]
[  3%] Built target OpenSSL_Target
Scanning dependencies of target platform
[  5%] Building C object src/core/CMakeFiles/core.dir/listener.c.o
[  7%] Building C object src/core/CMakeFiles/core.dir/library.c.o
[  8%] Building C object src/core/CMakeFiles/core.dir/lookup.c.o
[ 10%] Building C object src/platform/CMakeFiles/platform.dir/crypt.c.o
/home/dawnyi/workshop/test2/pynng-mqtt/nng/extern/msquic/src/core/library.c: In function ‘QuicLibrarySetGlobalParam’:
/home/dawnyi/workshop/test2/pynng-mqtt/nng/extern/msquic/src/core/library.c:896:10: error: ‘QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS’ undeclared (first use in this function); did you mean ‘QUIC_PARAM_GLOBAL_TEST_DATAPATH_HOOKS’?
  896 |     case QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS: {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          QUIC_PARAM_GLOBAL_TEST_DATAPATH_HOOKS
/home/dawnyi/workshop/test2/pynng-mqtt/nng/extern/msquic/src/core/library.c:896:10: note: each undeclared identifier is reported only once for each function it appears in
/home/dawnyi/workshop/test2/pynng-mqtt/nng/extern/msquic/src/core/library.c: In function ‘QuicLibraryGetGlobalParam’:
/home/dawnyi/workshop/test2/pynng-mqtt/nng/extern/msquic/src/core/library.c:1195:10: error: ‘QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS’ undeclared (first use in this function); did you mean ‘QUIC_PARAM_GLOBAL_TEST_DATAPATH_HOOKS’?
 1195 |     case QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS:
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          QUIC_PARAM_GLOBAL_TEST_DATAPATH_HOOKS
make[2]: *** [src/core/CMakeFiles/core.dir/build.make:206：src/core/CMakeFiles/core.dir/library.c.o] 错误 1
make[2]: *** 正在等待未完成的任务....
[ 12%] Building C object src/platform/CMakeFiles/platform.dir/hashtable.c.o
[ 14%] Building C object src/platform/CMakeFiles/platform.dir/pcp.c.o
[ 16%] Building C object src/platform/CMakeFiles/platform.dir/platform_worker.c.o
[ 17%] Building C object src/platform/CMakeFiles/platform.dir/toeplitz.c.o
make[1]: *** [CMakeFiles/Makefile2:226：src/core/CMakeFiles/core.dir/all] 错误 2
make[1]: *** 正在等待未完成的任务....
[ 19%] Building C object src/platform/CMakeFiles/platform.dir/inline.c.o
[ 21%] Building C object src/platform/CMakeFiles/platform.dir/platform_posix.c.o
[ 23%] Building C object src/platform/CMakeFiles/platform.dir/storage_posix.c.o
[ 25%] Building C object src/platform/CMakeFiles/platform.dir/datapath_epoll.c.o
[ 26%] Building C object src/platform/CMakeFiles/platform.dir/tls_openssl.c.o
[ 28%] Building C object src/platform/CMakeFiles/platform.dir/crypt_openssl.c.o
[ 30%] Building C object src/platform/CMakeFiles/platform.dir/certificates_posix.c.o
[ 32%] Building C object src/platform/CMakeFiles/platform.dir/selfsign_openssl.c.o
[ 33%] Linking C static library ../../obj/Release/libplatform.a
[ 35%] Built target platform
make: *** [Makefile:141：all] 错误 2

**Comment by nibanks** [comment]
I'm not sure. It seems like you might somehow be missing a C define to expose that type. But I don't know how that'd be possible.

**Reply by Dawn369** [reply]
Thanks for my answer, “QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS”， this variable can not find in /src/inc/msquic.h",where is it?
